### PR TITLE
Support for building with Java 9+

### PR DESCRIPTION
**Pull request by dapp**
I've removed references to classes that were made internal in Java 9, so that the compiler module can build using Java 11 (or Java 9+)

I've also removed the ComponentViewBinder interface and annotation, which was made obsolete by both Java lambdas and Kotlin behaviors.  I don't think any consumer ever used that feature, so even though it's a breaking change, I'd like to make this a 2.1.0 release...

Also, updated some view type names for consistency.

### CODE REVIEW

File: compiler/build.gradle [hunk 1]
```diff
@@ -10,9 +10,11 @@ dependencies {
   implementation project(path: ':blueprint-annotations')
   implementation 'com.google.auto.service:auto-service:1.0-rc3'
   implementation 'com.squareup:javapoet:1.9.0'
-  implementation files("${System.properties['java.home']}/../lib/tools.jar")
+//  implementation files("${System.properties['java.home']}/../lib/tools.jar")
   implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
   implementation 'androidx.recyclerview:recyclerview:1.0.0'
+  implementation 'org.apache.commons:commons-lang3:3.0'
+  runtimeOnly files('/Library/Java/JavaVirtualMachines/jdk1.8.0_311.jdk/Contents/Home/lib/tools.jar')
```
Comment (fkose696) [inline compiler/build.gradle:17]: I think this was added for test purposes? I don't think we can specify this for the repo. We might need to revert line 13

File: compiler/gradle.properties [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+#signing.password=
+#signing.secretKeyRingFile=
+
+android.injected.testOnly=false
+
+sonatypeUsername=
+sonatypePassword=
+android.useAndroidX=true
+android.enableJetifier=true
+org.gradle.java.home=/Library/Java/JavaVirtualMachines/jdk1.8.0_311.jdk/Contents/Home
```
Comment (fkose696) [inline compiler/gradle.properties:31]: same comment

File: gradle.properties [hunk 3]
```diff
@@ -28,3 +28,4 @@ sonatypeUsername=
 sonatypePassword=
 android.useAndroidX=true
 android.enableJetifier=true
+org.gradle.java.home=/Library/Java/JavaVirtualMachines/jdk1.8.0_311.jdk/Contents/Home
```
Comment (fkose696) [inline gradle.properties:31]: can you remove this?

File: compiler/build.gradle (hunk 1 above)
Comment (dapp) [reply compiler/build.gradle]: Done

File: compiler/gradle.properties (hunk 2 above)
Comment (dapp) [reply compiler/gradle.properties]: Done

File: gradle.properties (hunk 3 above)
Comment (dapp) [reply gradle.properties]: Done

**Review by fkose696 — APPROVED** [review]

**Pull request merged**